server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java

```diff
@@ -321,13 +330,23 @@ private String getRemoteAddr() {
         }
     }
 
-    private List<MailAddress> getSenderAsList(Mail originalMail) {
-        MaybeSender reversePath = originalMail.getMaybeSender();
+    private List<MailAddress> getSenderAsList(Mail originalMail) throws MessagingException {
+        Optional<MailAddress> returnPath = Optional.ofNullable(originalMail.getMessage().getHeader(RETURN_PATH))
+                .flatMap(path -> path.length > 0 ? Optional.ofNullable(path[0]) : Optional.empty())
```

Comment:
Let me suggest `Arrays.stream(path).findFirst()` that allows getting rid of the condition?

Reply:
I think we are far from other as mail address header parsing is complex...

Do we handle the following cases?

```
Return-Path: inv@lid@domain.tld
```

(I expect a fallback on the sender)

```
Return-Path: <wrapped@domain.tld>
```

(Eg mails from JIRA follows this pattern)

```
Return-Path: "Wrapped" <wrapped@domain.tld>
```

```
Return-Path: "Wrapped" 
    <wrapped@domain.tld>
```

& also encoding handling...